### PR TITLE
fix: reset insight on tab change

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

#14779 turned into a pretty chunky PR which had a fix hidden in it. Let's separate the fix.

## Changes

* resets insightLogic when changing editor tab in data exploration mode

This avoids racy behaviour where filter-based and query-based insights clash with each other for control of the UI

## How did you test this code?

👀 locally

### CODE REVIEW

**Comment by pauldambra** [comment]
This definitely fixes the tab switching problem for me @thmsobrmlr 

(whether it's the right fix is another question 😓)

**Comment by pauldambra** [comment]
Although _really_ I should follow your good example and wrap this in tests since we'll just break it again with the next change 🙈

**Comment by thmsobrmlr** [comment]
Unfortunately it's still broken for me 🤔 

https://user-images.githubusercontent.com/1851359/225886684-bc644cfd-f8cc-4a4f-b373-fb009f73dd8a.mov

File: cypress/e2e/insights.dataExploration.cy.ts [hunk 1]
```diff
@@ -26,9 +27,175 @@ describe('Insights (with data exploration on)', () => {
...
+}`)
+
+            cy.wait('@query').then(() => {
+                cy.get('tr.DataTable__row').should('have.length', 1)
+            })
+        })
+    })
+
+    describe('opening a new insight after opening a new SQL insight', () => {
+        // ugh, these tests have identical assertions to the ones above, but we need to open a SQL insight first
```
Comment (pauldambra) [inline cypress/e2e/insights.dataExploration.cy.ts:102]:
```suggestion
        // TRICKY: these tests have identical assertions to the ones above, but we need to open a SQL insight first
```

**Review by mariusandra — APPROVED** [review]
🤞 tests pass.. ✅ otherwise!

**Pull request merged**